plugins/generic/htmlArticleGalley/HtmlArticleGalleyPlugin.inc.php

```diff
@@ -153,7 +153,7 @@ protected function _getHTMLContents($request, $galley) {
 			if (!$referredArticle || $referredArticle->getId() != $submissionId) {
 				$referredArticle = $submissionDao->getById($submissionId);
 			}
-			$fileUrl = $request->url(null, 'article', 'download', array($referredArticle->getBestId(), $galley->getBestGalleyId(), $embeddableFile->getId()), $params);
+			$fileUrl = $request->url(null, 'article', 'download', array($referredArticle->getBestId(), $galley->getBestGalleyId(), $embeddableFile->getId(), $embeddableFile->getLocalizedData('name')));
```

Comment:
@asmecher `$embeddableFile->getLocalizedData('name')` is user input. Any chance of that breaking or being a security risk?

Reply:
No, I can't think of any.